src/lib/date.ts

```diff
@@ -49,3 +49,20 @@ export const getDisplayDuration = (duration: Duration): string => {
   }
   return sinceDateString
 }
+
+const SemesterToString = (semester: Semester): string => {
+  return semester === Semester.first ? '前期' : '後期'
+}
+
+export const getGroupOrProjectDurations = (
+  durations: YearWithSemesterDuration[]
+): string[] =>
+  durations.map(val => {
+    const since = `${val.since.year}${SemesterToString(val.since.semester)} ~`
+    if (val.until === undefined) {
+      return since
+    } else {
+      const until = `${val.until.year}${SemesterToString(val.until.semester)}`
+      return `${since} ${until}`
+    }
```

Comment:
```suggestion
    }

    const until = `${val.until.year}${SemesterToString(val.until.semester)}`
    return `${since} ${until}`
```
こうしちゃってよさそうです